src/request.rs

```diff
@@ -277,6 +277,47 @@ impl RequestBuilder {
         Ok(self)
     }
 
+    /// Sends a multipart/formdata body.
+    ///
+    /// ```no_run
+    /// use reqwest::mime;
+    /// use reqwest::{MultipartRequest, MultipartField, to_multipart};
+    ///
+    /// let client = reqwest::Client::new().unwrap();
+    /// let response = client.post("your url").unwrap()
+    ///     .multipart(
+    ///         // Add fields from anything serializable
+    ///         to_multipart(&[("key", "value"), ("key2", "value2")]).unwrap()
+    ///         // Add fields builder style
+    ///         .field(MultipartField::param("key3", "value3"))
+    ///         .field(MultipartField::param("json", "{ \"number\": 5 }")
+    ///             .mime(Some(mime::APPLICATION_JSON)))
+    ///         .field(MultipartField::file("file", "/path/to/file")
+    ///             .expect("File could not be opened"))
+    ///     ).send();
+    ///
+    /// ```
+    ///
+    /// See [`to_multipart`](fn.to_multipart.html), [`MultipartRequest`](struct.MultipartRequest.html)
+    /// and [`MultipartField`](struct.MultipartField.html) for more examples.
+    pub fn multipart(&mut self, multipart: MultipartRequest) -> &mut RequestBuilder {
```

Comment:
So, if a user has made use of a `Serialize` thing, I suspect they wouldn't need to use the builder to add more fields. They could have added them to their previous thing, right?

And for customizing parameters by using `MultipartField`, likewise, those could be just be added to a thing that implements `Serialize`, right?

```rust
let bike = vec![
    MultipartField::param("gears", "21"),
    MultipartField::param("color", "blue"),
    MultipartField::file("photo", "/usr/foo/photo.png")
        .mime(mime::IMAGE_PNG),
];

client.post(url)?
    .multipart(bike)?
    .send()?
```

Could something like this work? I should do another look at `ser.rs`...